src/components/Articles.vue

```diff
@@ -49,13 +52,7 @@ export default {
   }),
   computed: {
     articlesData: function() {
-      return this.articles.map(article => ({
-        title: article.title.value,
-        imageLink: article.teaserImage.value[0].url,
-        postDate: this.formatDate(article.postDate.value),
-        summary: article.summary.value,
-        link: `/${this.language}/articles/${article.system.id}`
-      }));
+      return this.articles.map(mapArticle, this.formatDate.bind(this), this.$t.bind(this));
```

Comment:
The syntax you are using is passing the arguments different way, I would prefer this syntax:

```suggestion
      return this.articles.map(article => mapArticle(article, this.formatDate.bind(this), this.$i18n.t.bind(this)));
```

Unfortunately, it seems it is not possible to use translation out of the component file, beacuse of this issue probably:
* https://github.com/kazupon/vue-i18n/issues/259

So let's map the article separately in Articles and Article component. Here is a sample hot to do it in Articles component.

```suggestion
      return this.articles.map(article => ({
        title: _.get(article, "title.value", this.$i18n.t("Articles.noTitleValue")),
        imageLink: _.get(article, "teaserImage.value[0].url"),
        postDate: this.formatDate(_.get(article, "postDate.value")),
        summary: _.get(article, "summary.value", this.$i18n.t("Articles.noSummaryValue")),
        link: `/${this.language}/articles/${_.get(article, "system.id")}`
      }));
```

Reply:
looking at https://github.com/kazupon/vue-i18n/issues/259, the issue stems out of the author not binding a context for `this` to his reference to `$t`, which this PR does do -- I wouldn't expect to be able to do:
```js
var fn = this.$t;
fn("some string");
```
the call to `fn` _should_ reasonably break as it's missing the context and could well refer to `this` within the function.

So the question comes down to which one is preferable:
1. local mappings in each consumer component, with duplicated code and no context binding
2. consolidation of common code in the `mapArticle` function, with bound functions passed in for date formatting & translation

I've tested with the latter, using a Kentico Cloud account, and my last updates work nicely, so it's down to the preference of style for this project. Thoughts?

Reply:
Just wondering about the style preference here:

1. localised, repeated code vs
2. DRY code with bound functions

each have their merits -- the final style choice is up to a project owner (:

Reply:
Sorry, for the delay @fluffynuts.
I would prefer local mapping - since the "viewmodel"  is a bit different for Article and Articles - one is using `summary` and the other `body_copy` element.

Reply:
@Simply007 no worries -- I've pushed a commit to address this (: